File: llmops/api/app/schemas/analysis.py
from pydantic import BaseModel, Field


class TrendData(BaseModel):
    x_axis: list[int]
    y_axis: list[float]


class IndicatorData(BaseModel):
    data: float
    pop: float


class AppAnalysisResponse(BaseModel):
    total_messages_trend: TrendData
    active_accounts_trend: TrendData
    avg_of_conversation_messages_trend: TrendData
    cost_consumption_trend: TrendData
    total_messages: IndicatorData
    active_accounts: IndicatorData
    avg_of_conversation_messages: IndicatorData
    token_output_rate: IndicatorData
    cost_consumption: IndicatorData


class GetAppAgentRuntimeAnalysisRequest(BaseModel):
    from_ts: int | None = Field(None, ge=0)
    to_ts: int | None = Field(None, ge=0)
    status: str = "all"
    user_id: str = "all"
    router_agent_id: str = "all"
    worker_agent_id: str = "all"
    group_by: str = "day"

File: llmops/ui/src/models/analysis.ts
import type { BaseResponse } from '@/models/base'
import type { AgentTaskRuntimeMetrics } from '@/models/agent-task'

// 获取应用分析结果响应结构
export type GetAppAnalysisResponse = BaseResponse<{
  total_messages: { data: number; pop: number }
  active_accounts: { data: number; pop: number }
  avg_of_conversation_messages: { data: number; pop: number }
  token_output_rate: { data: number; pop: number }
  cost_consumption: { data: number; pop: number }
  total_messages_trend: { x_axis: number[]; y_axis: number[] }
  active_accounts_trend: { x_axis: number[]; y_axis: number[] }
  avg_of_conversation_messages_trend: { x_axis: number[]; y_axis: number[] }
  cost_consumption_trend: { x_axis: number[]; y_axis: number[] }
}>

export type GetAppAgentRuntimeAnalysisRequest = {
  from_ts?: number
  to_ts?: number
  status?: string
  user_id?: string
  router_agent_id?: string
  worker_agent_id?: string
  group_by?: 'day' | 'worker' | 'router' | 'status' | string
}

export type GetAppAgentRuntimeAnalysisResponse = BaseResponse<AgentTaskRuntimeMetrics>
